### PR TITLE
Added 'defaults' table called 'preferences'

### PR DESCRIPTION
**Pull request by theopolis**
`select * from preferences` is the equivalent to OS X `defaults` without setting a domain (application ID) or plist path to infer a domain from. There's quite a bit of data and running it as a user may expose potentially sensitive data. Schedule this as ROOT only!

### CODE REVIEW

File: osquery/tables/system/darwin/preferences.cpp [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                            QueryData& results);
+
+void genOSXHashPreferences(const CFDictionaryRef& hash,
+                           const Row& base,
+                           QueryData& results) {
+  CFIndex count = CFDictionaryGetCount(hash);
+
+  // Allocate memory for copies of each key/value in dictionary.
+  auto keys = (void**)malloc(sizeof(void*) * count);
+  auto values = (void**)malloc(sizeof(void*) * count);
```
Comment (jedi22) [inline osquery/tables/system/darwin/preferences.cpp]: Is there a way we can have more this more strongly typed?

File: osquery/tables/system/darwin/preferences.cpp [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    r["value"] =
+        (CFBooleanGetValue((CFBooleanRef)value) == TRUE) ? "true" : "false";
+  } else if (CFGetTypeID(value) == CFDataGetTypeID()) {
+    // Do not include data preferences.
+  } else if (CFGetTypeID(value) == CFArrayGetTypeID()) {
+    genOSXListPreferences((CFArrayRef)value, base, results);
+    return;
+  } else if (CFGetTypeID(value) == CFDictionaryGetTypeID()) {
+    // Generate a row for each hash key.
+    genOSXHashPreferences((CFDictionaryRef)value, base, results);
```
Comment (jedi22) [inline osquery/tables/system/darwin/preferences.cpp]: Are stack overflows a concern here?

File: osquery/tables/system/darwin/preferences.cpp (hunk 1 above)
Comment (theopolis) [inline osquery/tables/system/darwin/preferences.cpp]: Unfortunately, not as far as I am aware, https://developer.apple.com/library/ios/documentation/CoreFoundation/Reference/CFDictionaryRef/index.html#//apple_ref/c/func/CFDictionaryGetKeysAndValues

File: osquery/tables/system/darwin/preferences.cpp (hunk 2 above)
Comment (theopolis) [inline osquery/tables/system/darwin/preferences.cpp]: A recursion limit would be nice, what do you think 10 or 20? After that I'd call foul, in practice the deepest I've seen is 5.

Comment (jedi22) [inline osquery/tables/system/darwin/preferences.cpp]: People like to do weird stuff, so I would say if you think the max is 5, give it 10. The stack can handle 10 and we should be covered in even exceptional cases.

**Comment by theopolis** [comment]
I'll add paths as an optional "trigger" to mimic the defaults command completely as another PR.

**Comment by theopolis** [comment]
Jenkins crashed on Centos, ok to test

**Pull request merged**